src/data/VariantSelector.ts

```diff
@@ -167,6 +168,49 @@ export function isValidPangoLineageQuery(query: string): boolean {
   return /^([A-Z]){1,3}(\.[0-9]{1,3})*(\.?\*)?$/.test(query.toUpperCase());
 }
 
+const stripNumber = (str: string) => Number.parseInt(str.replace(/\D+/g, ''));
+const isNumeric = (str: string) => /^\d+$/.test(str);
+
+type geneType = {
+  name: string;
+  startPosition: number;
+  endPosition: number;
+  aaSeq: string;
+};
+
+const formatGeneName = (gene: string): string => {
+  return gene.length === 1
+    ? gene.toUpperCase()
+    : gene.slice(0, -1).toUpperCase() + gene.slice(-1).toLowerCase();
+};
+
+const loadRefData = () => JSON.parse(JSON.stringify(jsonRefData));
```

Comment:
Could you please explain this function? Why first stringify and then parse it again?